students/nDruP/lesson09/mailroom.py

```diff
@@ -0,0 +1,223 @@
+    """
+    try:
+        conv_yes = conv_type(conv_str)
+        return conv_yes
+    except:
+        return None
+
+
+def input_donor_name(donor_name="list"):
+    """
+    Prompt user for donor name
+    Prompt->"list": show a list of the donor names
+    """
+    while donor_name.lower() == "list":
+        print(divider)
+        print("Enter Name (Pull up the list of donors by entering 'list')")
+        donor_name = user_input()
+        if donor_name.lower() == "list":
+            print(divider + d.names_str())
+    return donor_name
+
+
+def input_donor_amt(d_amt=0):
+    """
+    Prompt user for donation amount
+    If input cannot be converted to float, prompt again.
+    """
+    while True:
+        print("\nEnter a nonzero Donation Amount:")
+        d_amt = user_input()
+        if not d_amt:
+            break
+        d_amt = conv_str(d_amt, float)
+        if d_amt:
+            break
+        print("Enter a valid amount")
+    return d_amt
+
+
+def create_thank_u():
+    """
+    Compose and print a thank you letter to the donor for their donation.
+    Return to main
+    """
+    print(divider)
+    print("Let's craft a very personal thank you note for our donor!")
+
+    d_name = input_donor_name()
+    if d_name:
+       gift_amt = input_donor_amt()
+       if gift_amt:
+           d.add_donor(d_name, gift_amt)
+           thanks = d[d_name.lower()].thank_u_letter_str(1) 
+           print(thanks)
+           print(save_to_dir(d_name, thanks))
+    return
+
+
+def sort_report_by(report_sort=None):
+    """
+    Prompt user to determine how to display donor report.
+    """
+    while report_sort not in ['1', '2', '3', '4', '5']:
+        print(divider)
+        print("How would you like to sort the report?")
+        print("[1] Name Asc.\n[2] # of Gifts Desc.")
+        print("[3] Avg Gift Amount Desc.\n[4] Total Gift Amount Desc.")
+        print("[5] Unsorted")
+        report_sort = user_input()
+        if not report_sort:
+            return False
+    return int(report_sort)
+
+
+def create_donor_report():
+    """
+    Print a list of donors sorted by method chosen in sort_report_by.
+    Donor Name, Num Gifts, Average Gift, Total Given
+    """
+    sort_by = sort_report_by()
+    if sort_by:
+        report = d.donor_report(sort_by)
+        print(divider + report + divider)
+        print(save_to_dir("donor_report", report))
+    return
+
+
+def write_letters_to_all(write_dir = ""):
+    """
+    Write a full set of letters to each donor to individual files on disk.
+    Go through all donors in donor_dict, generate a thank you letter,
+    write to disk as a text file.
+    """
+    write_dir = input_dir()
+    if write_dir:
+        for donor in d.keys:
+            print(write_txt_to_dir(d[donor].name,
+                                   d[donor].thank_u_letter_str(),
+                                   write_dir))
+        print("Finished writing the letters")
+    return
+
+
+def exit():
```

Comment:
This overwrites the Python built-in `exit` function, so I'd suggest naming it something else